components/formats-gpl/utils/FileWriteSPW.java

```diff
@@ -0,0 +1,452 @@
+   *
+   * @param omexml the IMetadata object that is to be associated with the writer
+   * @return true if the file writer was successfully initialized; false if an
+   *   error occurred
+   */
+  private boolean initializeWriter(IMetadata omexml) {
+    // create the file writer and associate the OME-XML metadata with it
+    writer = new ImageWriter();
+    writer.setMetadataRetrieve(omexml);
+
+    Exception exception = null;
+    try {
+      writer.setId(outputFile);
+    }
+    catch (FormatException | IOException e) {
+      exception = e;
+    }
+    if (exception != null) {
+      System.err.println("Failed to initialize file writer.");
+    }
+    return exception == null;
+  }
+
+  /**
+   * Populate the minimum amount of metadata required to export a Plate.
+   *
+   */
+  private IMetadata initializeMetadata(int[][] nFovs) {
+    Exception exception = null;
+    try {
+      // create the OME-XML metadata storage object
+      ServiceFactory factory = new ServiceFactory();
+      service = factory.getInstance(OMEXMLService.class);
+      OMEXMLMetadata meta = service.createOMEXMLMetadata();
+      //IMetadata meta = service.createOMEXMLMetadata();
+      meta.createRoot();
+    
+      int plateIndex = 0;
+      int series = 0;     // count of images
+      int well = 0;
+  
+      meta.setPlateDescription(plateDescription,0); 
+ 
+      meta.setPlateID(MetadataTools.createLSID("Plate", 0), 0);
+      
+      meta.setPlateRowNamingConvention(NamingConvention.LETTER, 0);
+      meta.setPlateColumnNamingConvention(NamingConvention.NUMBER, 0);
+      meta.setPlateRows(new PositiveInteger(rows), 0);
+      meta.setPlateColumns(new PositiveInteger(cols), 0);
+      meta.setPlateName("First test Plate", 0);
+      
+      PositiveInteger pwidth = new PositiveInteger(width);
+      PositiveInteger pheight = new PositiveInteger(height);
+        
+      char rowChar = 'A';
+      for (int row = 0; row  < rows; row++) {
+        for (int column = 0; column < cols; column++) {
+          
+          // set up well
+          String wellID = MetadataTools.createLSID("Well:", well);
+          meta.setWellID(wellID, plateIndex, well);
+          meta.setWellRow(new NonNegativeInteger(row), plateIndex, well);
+          meta.setWellColumn(new NonNegativeInteger(column), plateIndex, well); 
+          
+          int nFOV= nFovs[row][column];
+          
+          for(int fov = 0; fov < nFOV ; fov++)  {
+            
+            // Create Image NB numberng in the Name goes from 1->n not 0-> n-1
+            String imageName = rowChar + ":" + Integer.toString(column + 1) + ":FOV:" + Integer.toString(fov + 1);
+            String imageID = MetadataTools.createLSID("Image", well, fov);
+            meta.setImageID(imageID, series);
+            meta.setImageName(imageName, series);
+            
+            String pixelsID = MetadataTools.createLSID("Pixels",row, well, fov);
+            meta.setPixelsID(pixelsID, series);
```

Comment:
The indexes in the Pixels ID are generally meant to match those in the Image ID - can you clarify or remove the ```row``` index here?